messaging/src/main/java/org/axonframework/eventhandling/gateway/ProcessingContextEventAppender.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventhandling.gateway;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.common.infra.ComponentDescriptor;
+import org.axonframework.common.infra.DescribableComponent;
+import org.axonframework.configuration.NewConfiguration;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.eventhandling.EventSink;
+import org.axonframework.messaging.MessageTypeResolver;
+import org.axonframework.messaging.unitofwork.ProcessingContext;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * Component that publishes events to an {@link EventSink} in the context of a {@link ProcessingContext}. The events
+ * will be published in the context this appender was created for. You can construct one through the
+ * {@link EventAppender#forContext(ProcessingContext, NewConfiguration)} method.
+ *
+ * @author Mitchell Herrijgers
+ * @since 5.0.0
+ */
+public class ProcessingContextEventAppender implements EventAppender, DescribableComponent {
```

Comment:
Nit: I assume we can make this class package-private, right?

Reply:
Yep, we can. Adjusting